### PR TITLE
display a message when there are no articles in a category

### LINKED ISSUE #202: feat: gracefully handle no results on category index page

### PR DESCRIPTION
**Pull request by mfrisbey**
Fix #202 

Test URLs:
- Before: https://main--petplace--hlxsites.hlx.page/article/category/pet-health/
- After: https://issue-202--petplace--hlxsites.hlx.page/article/category/pet-health/

Displays the message `No Articles Found` as an `h2` on the category index page if there are no articles in the category.

### CODE REVIEW

**Review by ramboz — APPROVED** [review]

File: templates/category-index/category-index.js [hunk 1]
```diff
@@ -42,25 +42,34 @@ async function renderArticles(articles) {
...
+  }
 }
 
 async function getArticles() {
   const categories = await getCategories();
   const applicableCategories = categories
-    .filter((c) => c.Path === window.location.pathname
-      || c['Parent Path'].startsWith(window.location.pathname))
+    .filter((c) => c.Path === 'window.location.pathname'
+      || c['Parent Path'].startsWith('window.location.pathname'))
```
Comment (ramboz) [inline templates/category-index/category-index.js:72]: That is probably not an intended change 🤔

File: templates/category-index/category-index.js [hunk 2]
```diff
@@ -42,25 +42,34 @@ async function renderArticles(articles) {
...
     const div = document.createElement('div');
     div.dataset.json = JSON.stringify(article);
+    articleCount += 1;
     meteringPromise = meterCalls(() => block.append(div));
   }
   document.querySelector('.pagination').dataset.total = articleLoadingPromise.total();
   await (meteringPromise || Promise.resolve());
   window.requestAnimationFrame(() => {
     block.querySelectorAll('.skeleton').forEach((sk) => sk.parentElement.remove());
   });
```
Comment (ramboz) [inline templates/category-index/category-index.js:58]: You can probably move this down after your if statement and drop L64 which does the same

File: templates/category-index/category-index.js (hunk 1 above)
Comment (mfrisbey) [reply templates/category-index/category-index.js:72]: Oops, yeah sorry. I'm still playing with it a bit. I've noticed that the "Page: 1 of ..." text still appears at the bottom, so I'm trying to figure out how to hide that, too.

Comment (ramboz) [reply templates/category-index/category-index.js:72]: you can hide the pagination block, I think

**Review by ramboz — APPROVED** [review]

**Pull request merged**